qiskit/utils/deprecation.py

```diff
@@ -37,15 +38,27 @@ def deprecate_arguments(
         Callable: The decorated callable.
     """
 
-    del since  # Will be used in a followup to add deprecations to our docs site.
-
     def decorator(func):
+        func_name = func.__name__
+        old_kwarg_to_msg = {}
+        for old_arg, new_arg in kwarg_map.items():
+            msg_suffix = (
+                "will in the future be removed." if new_arg is None else f"replaced with {new_arg}."
+            )
+            old_kwarg_to_msg[
+                old_arg
+            ] = f"{func_name} keyword argument {old_arg} is deprecated and {msg_suffix}"
```

Comment:
This logic needs to live here, rather than the helper function `_rename_kwargs`, because we need to call `_DeprecationMetadataEntry().store_on_function()` on the `wrapper` rather than the inner `func`